xlml/utils/gpu.py

```diff
@@ -109,12 +113,30 @@ def generate_gpu_name() -> str:
   return f"gpu-{str(uuid.uuid4())}"
 
 
+@task(provide_context=True)
+def select_zone(gcp: gcp_config.GCPConfig) -> airflow.XComArg:
+  context = get_current_context()
+  try_count = context["task_instance"].try_number - 1
```

Comment:
Do we need to specify try_number based on zone? i.e. we have 4 zones to try, however, the default try_number is 2.

Reply:
My understanding is that the `context["task_instance"].try_number` will be automatically  +1 updated everytime we  retry the task.